rimage/transform/parallel_projection_test.go

```diff
@@ -0,0 +1,135 @@
+package transform
+
+import (
+	"fmt"
+	"image"
+	"os"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/test"
+	"go.viam.com/utils/artifact"
+
+	pc "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// Helper function that makes and returns a PointCloud from an artifact path consisting of the first numPoints of points.
+func makePointCloudFromArtifact(t *testing.T, artifactPath string, numPoints int) (pc.PointCloud, error) {
+	t.Helper()
+	pcdFile, err := os.Open(artifact.MustPath(artifactPath))
+	if err != nil {
+		return nil, err
+	}
+	pcd, err := pc.ReadPCD(pcdFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if numPoints == 0 {
+		return pcd, nil
+	}
+
+	shortenedPC := pc.NewWithPrealloc(numPoints)
+
+	counter := numPoints
+	pcd.Iterate(0, 0, func(p r3.Vector, d pc.Data) bool {
+		if counter > 0 {
+			err = shortenedPC.Set(p, d)
+			counter--
+		}
+		return err == nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return shortenedPC, nil
+}
+
+func TestParallelProjectionOntoXZWithRobotMarker(t *testing.T) {
+	t.Run("Project an empty pointcloud", func(t *testing.T) {
+		p := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
+		ppRM := NewParallelProjectionOntoXZWithRobotMarker(&p)
+
+		pointcloud := pc.New()
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(pointcloud)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "calculation of the X-coord's mean during pcd projection failed")
```

Comment:
[q] Does this fail because the pointcloud was empty? If so, could we return that as the error message to clarify the cause of the failure?

Reply:
I will clarify.

Reply:
Also, due to that fact golang doesn't handling overflows innately, most packages out there don't deal have them. I spoke to Eric about this and he echoed what aaron said on slack using 'big numbers'. However Eric was mentioned that they take up a lot of memory and for a significantly large pointcloud to allow us to worry about this, it might cause other issues unless done carefully.

I think we punt the overflow/underflow issue until it is a problem, or until viam develops its own stats package, but want to hear your thoughts